synapse/rest/media/v1/storage_provider.py

```diff
@@ -88,20 +89,30 @@ async def store_file(self, path, file_info):
             return None
 
         if self.store_synchronous:
-            return await self.backend.store_file(path, file_info)
+            # store_file is supposed to return an Awaitable, but guard
+            # against improper implementations.
+            result = self.backend.store_file(path, file_info)
+            if inspect.isawaitable(result):
+                return await result
         else:
             # TODO: Handle errors.
-            def store():
+            async def store():
                 try:
-                    return self.backend.store_file(path, file_info)
+                    result = self.backend.store_file(path, file_info)
+                    if inspect.isawaitable(result):
+                        return await result
```

Comment:
This isn't 100% necessary, but I think without it we're going to end up with "coroutine wasn't awaited on" in the logs.

Reply:
`coroutine wasn't awaited on` means that it won't get run at all, aiui.

I think in this case it was fine because the coroutine was returned to `run_in_background` which then awaits

Reply:
Ah fair enough, should I revert this then?

Reply:
Nah, tbh i think consistency here ftw